contracts/BondTellerErc20.sol

```diff
@@ -0,0 +1,464 @@
+     * @param name_ The name of the bond token.
+     * @param governance_ The address of the [governor](/docs/protocol/governance).
+     * @param solace_ The SOLACE token.
+     * @param xsolace_ The xSOLACE token.
+     * @param pool_ The underwriting pool.
+     * @param dao_ The DAO.
+     * @param principal_ address The ERC20 token that users deposit.
+     * @param bondDepo_ The bond depository.
+     */
+    function initialize(
+        string memory name_,
+        address governance_,
+        address solace_,
+        address xsolace_,
+        address pool_,
+        address dao_,
+        address principal_,
+        address bondDepo_
+    ) external override initializer {
+        __Governable_init(governance_);
+        require(solace_ != address(0x0), "zero address solace");
+        require(xsolace_ != address(0x0), "zero address xsolace");
+        require(pool_ != address(0x0), "zero address pool");
+        require(dao_ != address(0x0), "zero address dao");
+        require(principal_ != address(0x0), "zero address principal");
+        require(bondDepo_ != address(0x0), "zero address bond depo");
+        string memory symbol = "SBT";
+        __ERC721Enhanced_init(name_, symbol);
+        solace = ISOLACE(solace_);
+        xsolace = IxSOLACE(xsolace_);
+        solace.approve(xsolace_, type(uint256).max);
+        underwritingPool = pool_;
+        dao = dao_;
+        principal = IERC20(principal_);
+        bondDepo = IBondDepository(bondDepo_);
+    }
+
+    /***************************************
+    VIEW FUNCTIONS
+    ***************************************/
+
+    // BOND PRICE
+
+    /**
+     * @notice Calculate the current price of a bond.
+     * Assumes 1 SOLACE payout.
+     * @return price_ The price of the bond measured in `principal`.
+     */
+    function bondPrice() public view returns (uint256 price_) {
+        uint256 timeSinceLast = block.timestamp - lastPriceUpdate;
+        price_ = exponentialDecay(nextPrice, timeSinceLast);
+        if (price_ < minimumPrice) {
+            price_ = minimumPrice;
+        }
+    }
+
+    /**
+     * @notice Calculate the amount of **SOLACE** or **xSOLACE** out for an amount of `principal`.
+     * @param amountIn Amount of principal to deposit.
+     * @param stake True to stake, false to not stake.
+     * @return amountOut Amount of **SOLACE** or **xSOLACE** out.
+     */
+    function calculateAmountOut(
+        uint256 amountIn,
+        bool stake
+    ) external view returns (uint256 amountOut) {
+        // exchange rate
+        amountOut = 1 ether * amountIn / bondPrice(); // 1 ether => 1 solace
+        // ensure there is remaining capacity for bond
+        if (capacityIsPayout) {
+            // capacity in payout terms
+            require(capacity >= amountOut, "bond at capacity");
+        } else {
+            // capacity in principal terms
+            require(capacity >= amountIn, "bond at capacity");
+        }
```

Comment:
require(capacityIsPayout && (capacity >= amountOut), "bond at capacity");
require(capacity >= amountIn, "bond at capacity");

Not big deal, just saving for the space.